src/ripple/core/impl/Config.cpp

```diff
@@ -1017,6 +1017,37 @@ Config::loadFromString(std::string const& fileContents)
             }
         }
     }
+
+    // Add DatabasePinned validation
+    if (!RUN_STANDALONE)
+    {
+        auto db_section = section(ConfigSection::nodeDatabase());
+        if (db_section.exists("pinned_type"))
+        {
+            // Ensure base type is specified
+            if (!db_section.exists("type"))
+            {
+                Throw<std::runtime_error>(
+                    "type must be specified when using pinned_type");
+            }
+
+            // Ensure pinned_path is specified
+            if (!db_section.exists("pinned_path"))
+            {
+                Throw<std::runtime_error>(
+                    "pinned_path is required when pinned_type is set");
+            }
+
+            // Ensure online_delete is set for rotation
+            if (auto delete_interval = get(db_section, "online_delete", 0);
+                delete_interval == 0)
+            {
+                Throw<std::runtime_error>(
```

Comment:
It's not for memory, and it's more that it's pointless to use pinning without online_delete